include/seqan3/argument_parser/auxiliary.hpp

```diff
@@ -0,0 +1,145 @@
+ */
+
+#pragma once
+
+namespace seqan3
+{
+
+/*!\brief Used to further specify argument_parser options/flags.
+ *
+ * \details
+ * All options and flags are set to option_spec::DEFAULT unless specified
+ * otherwise by the developer, e.g. when calling argument_parser::add_option().
+ *
+ * ```cpp
+ *     myparser.add_option(myvar, 's', "special-op", "You know what you doin'?",
+ *                         option_spec::ADVANCED);
+ * ```
+ *
+ * \cond DEV
+ * The option_spec type is needed in all the format classes (format_parse,
+ * format_help, ...etc) and the argument_parser (in the declaration of the
+ * add_option/add_flag functions), so it can't be part of the
+ * argument_parser class itself but need to be independent.
+ * \endcond
+ */
+enum option_spec
+{
+    DEFAULT  = 0, //!< The default were no checking or special displaying is happening.
+    REQUIRED = 1, /*!< Set an option as required if you want to enforce that the user
+                   * supplies this option when calling the program via the command line.
+                   * If the option is missing, the argument_parser will automatically
+                   * detect this and throw a invalid_argument exeption.
+                   */
+    ADVANCED = 2, /* Set an option/flag to advanced if you do not want the option to
+                   * be displayed in the normal help page (`-h/--help`). Instead, the
+                   * advanced options are only displayed when calling `-hh/--advanced-help`
+                   */
+    HIDDEN   = 4, /* Set an option/flag to hidden, if you want to complete hide it from
+                   * the user. It will never appear on the help page nor any export format.
+                   * For example, this can be useful for debugging reasons.
+                   * (e.g. "A tool for mapping reads to the genome").
+                   */
+};
+
+/*!\brief Stores all parser related meta information of the seqan3::argument_parser.
+ *
+ * \attention You should supply as much information as possible to help the users
+ * of your application.
+ *
+ * \details
+ * The meta information is assembled in a struct to provide a central access
+ * point that can be easyily extended.
+ *
+ * \cond DEV
+ * The meta information are needed in all the format classes (format_parse,
+ * format_help, ...etc) and the argument_parser, so they can't be part of the
+ * argument_parser class itself but need to be independent.
+ * \endcond
+ */
+struct argument_parser_meta_data // holds all meta information
+{
+    //!\brief The application name that will be displayed on the help page.
+    std::string app_name;
+    //!\brief The version information `MAJOR.MINOR.PATH` (e.g. 3.1.3)
+    std::string version;
+    //!\brief A short description of the application (e.g. "A tool for mapping reads to the genome").
+    std::string short_description;
+    //!\brief Your name ;-)
+    std::string author;
+    //!\brief The authors email adress for correspondance.
```

Comment:
correspondence

Reply:
multiple issues: author's e-mail address

Reply:
please use a spell checker for British English in your editor :)